### ISSUE TITLE
Metric Dimensions - Custom Properties

### ISSUE THREAD

**Issue by dylan-asos** [jdvor/appmetrics-applicationinsights#6]
Your documentation states : 

> MetricTelemetry does not have a way how to explicitly report dimensions, so this reporter reports them as new metric records with names derived for the parent one.

However, multi dimensional metrics are supported by AI Metric Telemetry, by recording the data as custom properties - there's no need to create entire new metric names like 

> "fruit_count", "apples.apples" and "fruit_count.pears".

This would give you metric telemetry like
```
{
    "metric": "fruit_count",
    "time": "2019-04-01T11:31:00",
    "dimNames": [
      "FruitType"
    ],
    "series": [
      {
        "dimValues": [
          "apple"
        ],
        "min": 1,
        "max": 1,
        "sum": 1,
        "count": 1
      },
      {
        "dimValues": [
          "pear"
        ],
        "min": 1,
        "max": 1,
        "sum": 1,
        "count": 1
      }                    
    ]
 }
```

**Comment by jdvor** [comment]
Thanks for the tip. I'll check it out.

**Comment by jdvor** [comment]
I've released v2.0.0 where it is possible to choose how to deal with AppMetrics items. See section "AppMetrics item as Application Insights customDimension" of README.md for details.
```
var metrics = new MetricsBuilder()
    .Configuration.Configure(metricsOptions)
    .Report.ToApplicationInsights(opts => {
        opts.InstrumentationKey = "00000000-0000-0000-0000-000000000000";
        opts.ItemsAsCustomDimensions = true;
        opts.DefaultCustomDimensionName = "item";
    })
    .Build();
```

Original behavior where items are part of the customMetric name remains default.

**Comment by dylan-asos** [comment]
Great stuff @jdvor 👍
